c/tree.c

```diff
@@ -1,16 +1,236 @@
 #include "tree.h"
 
+#ifndef WIN32
+//  WIN32 build and link see https://github.com/witwall/mman-win32
 #include <sys/mman.h>
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
+
+#ifdef WIN32
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#include <_mingw.h>
+#include <windows.h>
+#include <errno.h>
+#include <io.h>
+#endif
+
 #include <fcntl.h>
 #include <errno.h>
 #include <inttypes.h>
 #include <netdb.h>
 #include <stdbool.h>
 #include <stdio.h>
 
+#ifdef WIN32
+	// Source: https://github.com/witwall/mman-win32
+
+	#define PROT_NONE       0
+	#define PROT_READ       1
+	#define PROT_WRITE      2
+	#define PROT_EXEC       4
+
+	#define MAP_FILE        0
+	#define MAP_SHARED      1
+	#define MAP_PRIVATE     2
+	#define MAP_TYPE        0xf
+	#define MAP_FIXED       0x10
+	#define MAP_ANONYMOUS   0x20
+	#define MAP_ANON        MAP_ANONYMOUS
+
+	#define MAP_FAILED      ((void *)-1)
+
+	/* Flags for msync. */
+	#define MS_ASYNC        1
+	#define MS_SYNC         2
+	#define MS_INVALIDATE   4
+
+	void*   mmap(void *addr, size_t len, int prot, int flags, int fildes, off_t off);
+	int     munmap(void *addr, size_t len);
+	int     mprotect(void *addr, size_t len, int prot);
+	int     msync(void *addr, size_t len, int flags);
+	int     mlock(const void *addr, size_t len);
+	int     munlock(const void *addr, size_t len);
+
+	#ifndef FILE_MAP_EXECUTE
+	#define FILE_MAP_EXECUTE    0x0020
+	#endif /* FILE_MAP_EXECUTE */
+
+	static int __map_mman_error(const DWORD err, const int deferr)
+	{
+		if (err == 0)
```

Comment:
After splitting out the code, please run `dev-bin/uncrustify-all.sh` on it so that it is formatted similarly to our other code.

Reply:
@oschwald can you provide some guidance for adding another file to the build list?

with regards to dev-bin/uncrustify-all.sh, I'm afraid my environment lacks the tools.